### ISSUE TITLE
[a11y]: FileUploaderItem - Invalid aria-described-by when not invalid

### ISSUE THREAD

**Issue by francinelucca** [carbon-design-system/carbon#14004]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.31.3

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Accessibility Checker, Latest Deployment

### Description

The aria-described-by id only exists in the DOM if the item is invalid, so we need to only add it when it is invalid.

<img width="773" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40550942/4836c59e-abba-468d-98bb-a46ea9e81374">
<img width="666" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40550942/60ed31ae-765b-45cb-84ff-115000af6dbb">

### WCAG 2.1 Violation

https://www.ibm.com/able/requirements/requirements/#4_1_2

### Reproduction/example

https://stackblitz.com/edit/github-g18ssf?file=src%2FApp.jsx

### Steps to reproduce

Run a11y checker on repro example, notice issue

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by andreancardona** [comment]
This issue is fixed on version @carbon/react 1.34.0 - marking as closed and resolved
